mindsdb/integrations/handlers/replicate_handler/replicate_handler.py

```diff
@@ -0,0 +1,143 @@
+import types
+from mindsdb.utilities.config import Config
+
+
+class ReplicateHandler(BaseMLEngine):
+    name = "replicate"
+
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+        if 'using' not in args:
+            raise Exception("Replicate engine requires a USING clause! Refer to its documentation for more details.")
+        else:
+            args = args['using']
+
+        if 'model_name' and 'version' not in args:
+            raise Exception('Add model_name and version ')
+
+            # Checking if passed model_name and version  are correct or not
+        try:
+            replicate.default_client.api_token = args['api_key']
+            replicate.models.get(args['model_name']).versions.get(args['version'])
+        except Exception as e:
+            raise Exception("Check your model_name and version carefully", e)
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
+        """Saves model details in stirage to access it later
+        """
+        args = args['using']
+        args['target'] = target
+        self.model_storage.json_set('args', args)
+
+    def predict(self, df: pd.DataFrame, args: Optional[Dict] = None) -> pd.DataFrame:
+        """Using replicate makes the prediction according to your parameters
+        """
+        def filter_images(x):
+            output = replicate.run(
+                    f"{args['model_name']}:{args['version']}",
+                    input={**x[1].to_dict(), **pred_args}) # unpacking parameters inputted
+
+            if isinstance(output, types.GeneratorType):
+                # getting final url if output is generator of frames url
+                output = [list(output)[-1]]
+                return output
+            else:
+                return output
+
+        pred_args = args['predict_params'] if args else {}
+        args = self.model_storage.json_get('args')
+        
+        # raiseing Exception if wrong parameters is given
+        params_names = set(df.columns) | set(pred_args)
+        available_params = self._get_schema(only_keys=True)
+        wrong_params = []
+        for i in params_names:
+            if i not in available_params:
+                wrong_params.append(i)
+
+        if wrong_params:
+            raise Exception(f"""
+'{wrong_params}' is/are not supported parameter for this model.
+Use DESCRIBE PREDICTOR mindsdb.<model_name>.features; to know about available parameters. OR 
+Visit https://replicate.com/f{args['model_name']}/versions/{args['version']} to check parameters.
+            """)
+
+        replicate.default_client.api_token = self._get_replicate_api_key(args)
+
+        rows = df.iterrows()      # Generator is returned
+        rows_arr = list(rows)      # Generator converted to list have tuple consisting (index, row(pandas.Series))
+
+        urls = map(filter_images, rows_arr)  # using filter_images function to get url according to inputted parameters
```

Comment:
This seems like a huge difference between what actually happens and what is disclaimed in the documentation (e.g. `README.md`). As far as I can understand, this means the handler right now would only support text to image models, correct?

If so, please rework the entire README and other documentation to _clearly_ state this.